ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/MetricsInterceptor.scala

```diff
@@ -79,12 +81,45 @@ final class MetricsInterceptor(metrics: MetricRegistry) extends ServerIntercepto
   override def interceptCall[ReqT, RespT](
       call: ServerCall[ReqT, RespT],
       headers: Metadata,
-      next: ServerCallHandler[ReqT, RespT]): ServerCall.Listener[ReqT] = {
+      next: ServerCallHandler[ReqT, RespT],
+  ): ServerCall.Listener[ReqT] = {
     val fullMethodName = call.getMethodDescriptor.getFullMethodName
     val metricName = fullServiceToMetricNameCache.getOrElseUpdate(
       fullMethodName,
       MetricsInterceptor.nameFor(fullMethodName))
-    metrics.meter(metricName).mark()
-    next.startCall(call, headers)
+    val timer = metrics.timer(metricName).time()
+    val listener = next.startCall(call, headers)
+    new TimedListener(listener, timer)
```

Comment:
How does this behave with streaming responses? Does it only stop the timer when the stream is closed?

Reply:
I believe so; it's really hard to tell and I couldn't find a good way to test it, but the numbers look about right.

Reply:
OK, wrote some tests; looks good.